lib/src/soloud.dart

```diff
@@ -625,6 +658,166 @@ interface class SoLoud {
+          sampleRate,
+          channels.count,
+          pcmFormat.value,
+          onBuffering,
+        );
+
+    final newSound = _addNewSound(ret.error, '', ret.soundHash.hash);
+    return newSound;
+  }
+
+  /// Add PCM audio data to the stream.
+  ///
+  /// This method can be called within an `Isolate` making possible
+  /// to create PCM data and send them to the buffer without frezing
+  /// the main thread.
+  /// When finishing to add data to the stream, call [setDataIsEnded].
+  ///
+  /// Throws [SoLoudNotInitializedException] if the engine is not initialized.
+  /// Throws [SoLoudPcmBufferFullOrStreamEndedCppException] if the buffer
+  /// is full or stream buffer has been set to be ended.
+  /// Thows [SoLoudOutOfMemoryException] if the buffer is out of OS memory or
+  /// the given `maxBufferSize` of the `setBufferStream` call is too small.
+  ///
+  /// See also:
+  ///
+  ///  * [addAudioDataStramF32] to add `Float32List` PCM audio data.
+  ///  * [addAudioDataStreamS8] to add `Int8List` PCM audio data.
+  ///  * [addAudioDataStreamS16] to add `Int16List` PCM audio data.
+  ///  * [addAudioDataStreamS32] to add `Int32List` PCM audio data.
+  void addAudioDataStreamU8(
+    AudioSource source,
+    Uint8List audioChunk,
+  ) {
+    if (!isInitialized) {
+      throw const SoLoudNotInitializedException();
+    }
+
+    final e = SoLoudController().soLoudFFI.addAudioDataStream(
+          source.soundHash.hash,
+          audioChunk,
+        );
+
+    if (e != PlayerErrors.noError) {
+      _logPlayerError(e, from: 'addAudioDataStream() result');
+      throw SoLoudCppException.fromPlayerError(e);
+    }
+  }
+
+  /// Helper method to add `Float32List` PCM audio data to the stream.
+  ///
+  /// This method is a wrapper around [addAudioDataStreamU8].
+  void addAudioDataStramF32(AudioSource source, Float32List audioChunk) {
+    addAudioDataStreamU8(source, audioChunk.buffer.asUint8List());
+  }
+
+  /// Helper method to add `Int8List` PCM audio data to the stream.
+  ///
+  /// This method is a wrapper around [addAudioDataStreamU8].
+  void addAudioDataStreamS8(AudioSource source, Int8List audioChunk) {
+    addAudioDataStreamU8(source, audioChunk.buffer.asUint8List());
+  }
+
+  /// Helper method to add `Int16List` PCM audio data to the stream.
+  ///
+  /// This method is a wrapper around [addAudioDataStreamU8].
+  void addAudioDataStreamS16(AudioSource source, Int16List audioChunk) {
+    addAudioDataStreamU8(source, audioChunk.buffer.asUint8List());
+  }
+
+  /// Helper method to add `Int32List` PCM audio data to the stream.
+  ///
+  /// This method is a wrapper around [addAudioDataStreamU8].
+  void addAudioDataStreamS32(AudioSource source, Int32List audioChunk) {
+    addAudioDataStreamU8(source, audioChunk.buffer.asUint8List());
+  }
+
+  /// Set the end of the data stream.
+  ///
+  /// By setting the stream to be ended means that when playing it, it will stop
+  /// at that point (if the loop is not set).
```

Comment:
What does it mean when the loop is set, though? Maybe add a paragraph saying what it means to call this function when `loop` is set.

Reply:
Yes right, my explanation is bad. Corrected to:
```
  /// By setting the stream to be ended means that when playing it, it can
  /// handle the stop event when it reaches the end of the data stream or can
  /// be looped if the looping is enabled.
```
Done.